src/ioctls/vcpu.rs

```diff
@@ -1568,6 +1574,17 @@ impl VcpuFd {
                 KVM_EXIT_HYPERV => Ok(VcpuExit::Hyperv),
                 r => Ok(VcpuExit::Unsupported(r)),
             }
+        } else if ret == -1 && errno::Error::last() == errno::Error::new(libc::EFAULT) {
+            let run = self.kvm_run_ptr.as_mut_ref();
+            match run.exit_reason {
+                KVM_EXIT_MEMORY_FAULT => {
+                    // SAFETY: Safe because the exit_reason (which comes from the kernel) told us
+                    // which union field to use.
+                    let fault = unsafe { &mut run.__bindgen_anon_1.memory_fault };
+                    Ok(VcpuExit::MemoryFault(fault.flags, fault.gpa, fault.size))
+                }
+                _ => Err(errno::Error::last()),
+            }
```

Comment:
I'd probably write this as 

```rs
else {
    let errno = errno::Error::last();
    if errno.0 == libc::EFAULT && ret == - 1 {
        let run = ...
    } else {
        Err(errno)
    }
}
```

to avoid multiple calls to `errno::Error::last()`

Reply:
Also, can we add a comment here about why this exit reason is so weird (its the only one that comes with a non-zero return value after all!)

Reply:
done